### PR TITLE
[l10n] Add Icelandic (is-IS) locale

### PR DESCRIPTION
**Pull request by magnimarels**
Added Icelandic translation for the x-data-grid

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

**Review by LukasTy — COMMENTED** [review]
Thank you for your contribution! 🙏 
Could you please sync (merge into) your branch with the latest `upstream/master` to avoid unwanted changes in the PR? 🤔

File: packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export interface GridAggregationFunction<V = any, AV = V> {
   getCellValue?: (params: GridAggregationGetCellValueParams) => V;
 }
 
-interface GridAggregationParams<V = any> {
+export interface GridAggregationParams<V = any> {
```
Comment (MBilalShafi) [inline packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts:77]: May I ask why is this export needed along with this PR, or do we need to rebase with the master branch?

Comment (magnimarels) [reply packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts:77]: Not sure why that's in there as I didn't write that line

Comment (MBilalShafi) [reply packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts:77]: Could you try merging the latest `master` branch?

Comment (LukasTy) [reply packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts:77]: Confirming that `upstream/master` should be merged into this branch. 😉

Comment (magnimarels) [reply packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts:77]: I just did that earlier, am I missing something that I'm doing wrong?

Comment (LukasTy) [reply packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts:77]: Are you sure that you merged `upstream/master` and not your out of date fork's master? 🤔

Comment (magnimarels) [reply packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts:77]: You're right, I missed that, should be updated now

Comment (magnimarels) [reply packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts:77]: It's all good, I can see where the misunderstanding came from. 
I followed the CONTRIBUTING.md which linked to the CONTRIBUTING.md from material-ui [https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md](url)
Which talks about using the next branch.

I'm not too familiar with rebasing but I'll try it, if that doesn't work out I'll open a new PR from the correct branch

Comment (LukasTy) [reply packages/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts]: @magnimarels I have rebased the branch onto the latest `upstream/master` and I've discarded the oldest commits which looked like duplicates. 😉 
Please double-check if everything looks correct. 🙏 
I'm leaving the final review to @mui/xgrid. 👍

**Comment by LukasTy** [comment]
🔔 Gentle ping @mui/xgrid.

**Review by MBilalShafi — APPROVED** [review]
Thank you for your contribution 👍

**Comment by LukasTy** [comment]
Forcing a merge as the e2e website seems to be flaky and would definitely pass. 👌

**Pull request merged**